cmd/newrelic-infra/newrelic-infra.go

```diff
@@ -228,7 +224,24 @@ func initializeAgentAndRun(c *config.Config, logFwCfg config.LogForward) error {
 	// Basic initialization of the agent.
 	timedLog.WithField("version", buildVersion).Info("Initializing")
 
-	agt, err := agent.NewAgent(c, buildVersion, ffManager)
+	registerClient, err := identityapi.NewIdentityRegisterClient(
+		c.IdentityURL,
+		c.License,
+		userAgent,
+		c.PayloadCompressionLevel,
+		httpClient,
+	)
+	if err != nil {
+		return err
+	}
+
+	agt, err := agent.NewAgent(
+		c,
+		buildVersion,
+		userAgent,
+		registerClient,
+		ffManager)
```

Comment:
`NewAgent` constructor is aimed to be called in the simplest way possible so caller (main fn) has less responsibilities, while its body is the proper place for building services and afterwards pass them into `agent.New` which is aimed to build agents injecting provided services (for testing and so).